File: marcs-cakewalk.js
function marcsCakewalk(calorie) {
    const descArr = calorie.sort((a, b) => b - a); 
    let miles = 0;
    for (let index = 0; index < descArr.length; index++) {
        miles += (Math.pow(2, index) * descArr[index]);
    }
    return miles;
}

File: marcs-cakewalk.kt
fun marcsCakewalk(calorie: Array<Int>): Long {
    var miles: Double = 0.0
    calorie.sort()
    calorie.reverse()
    for (index in 0..calorie.size - 1) {
        miles += (2.toDouble().pow(index) * calorie[index])
    }
    return miles.toLong()
}